x-pack/metricbeat/module/statsd/server/server.go

```diff
@@ -55,13 +91,72 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}
 
 	processor := newMetricProcessor(config.TTL)
+
+	mappings, err := buildMappings(config.Mappings)
+	if err != nil {
+		return nil, fmt.Errorf("invalid mapping configuration for `statsd.mapping`: %w", err)
+	}
 	return &MetricSet{
 		BaseMetricSet: base,
 		server:        svc,
+		mappings:      mappings,
 		processor:     processor,
 	}, nil
 }
 
+func buildMappings(config []StatsdMapping) (map[string]StatsdMapping, error) {
+	mappings := make(map[string]StatsdMapping, len(config))
+	for _, mapping := range config {
+		regexPattern := strings.Replace(mapping.Metric, ".", `\.`, -1)
+		regexPattern = strings.Replace(regexPattern, "<", "(?P<", -1)
+		regexPattern = strings.Replace(regexPattern, ">", ">[^.]+)", -1)
+		var err error
+		mapping.regex, err = regexp.Compile(fmt.Sprintf("^%s$", regexPattern))
+		if err != nil {
+			return nil, err
```

Comment:
Nit. Include in this error the failing pattern, so it is easier to find the culprit when one fails.
```suggestion
			return nil, fmt.Errorf("invalid pattern %s: %w", mapping.Metric, err)
```